ecosystem_tests/ecosystem_tests_cli/ecosystem_tests.py

```diff
@@ -0,0 +1,244 @@
+            content = licence_file.read()
+        return base64.b64encode(content.encode('utf-8')).decode('ascii')
+    elif value == DEFAULT_LICENSE_PATH:
+        raise EcosystemTestCliException(
+            'Liscence not found in default location: {path}'.format(
+                path=DEFAULT_LICENSE_PATH))
+    return value
+
+
+def secrets_callback(ctx, param, value):
+    """
+    Prepare secrets as base64 encoded string for dorkl
+    :return dictonary contains {secret_key:secret_base_64_encoded}
+    """
+    if not value or ctx.resilient_parsing:
+        return {}
+    return secrets_to_dict(value)
+
+
+def file_secrets_callback(ctx, param, value):
+    """Prepare secrets from file as base64 encoded string for dorkl
+    return dictonary contains {secret_key:file_content_base_64_encoded}
+    """
+    if not value or ctx.resilient_parsing:
+        return {}
+    return file_secrets_to_dict(value)
+
+
+def encoded_secrets_callback(ctx, param, value):
+    """Prepare encoded secrets for dorkl
+    :return dictonary contains {secret_key:secret_base_64_encoded}
+    """
+    if not value or ctx.resilient_parsing:
+        return {}
+    return encoded_secrets_to_dict(value)
+
+
+def plugins_callback(ctx, param, value):
+    if not value or ctx.resilient_parsing:
+        return []
+    return create_plugins_list(value)
+
+
+class Options(object):
+    def __init__(self):
+        """The options api is nicer when you use each option by calling
+        `@ecosystem_tests.options.some_option` instead of
+        `@ecosystem_tests.some_option`.
+        Note that some options are attributes and some are static methods.
+        The reason for that is that we want to be explicit regarding how
+        a developer sees an option. It it can receive arguments, it's a
+        method - if not, it's an attribute.
+        """
+
+        self.blueprint_path = click.option('-b',
+                                           '--blueprint-path',
+                                           default=[DEFAULT_BLUEPRINT_PATH],
+                                           type=click.Path(),
+                                           multiple=True,
+                                           show_default=DEFAULT_BLUEPRINT_PATH)
+
+        self.inputs = click.option(
+            '-i',
+            '--inputs',
+            multiple=True,
+            callback=inputs_callback,
+            type=click.STRING,
+            help=helptexts.INPUTS)
+
+        self.timeout = click.option('-t',
+                                    '--timeout',
+                                    type=int,
+                                    default=TIMEOUT,
+                                    help=helptexts.TEST_TIMEOUT,
+                                    show_default=TIMEOUT)
+        self.test_id = click.option('--test-id',
+                                    type=click.STRING,
+                                    help=helptexts.TEST_ID)
```

Comment:
What do you think about adjusting `validate_and_generate_test_ids` to happen here? That way we don't enter the test with invalid ID.